staker/bold/bold_staker.go

```diff
@@ -455,6 +514,15 @@ func newBOLDChallengeManager(
 		apiAddr := fmt.Sprintf("%s:%d", config.APIHost, config.APIPort)
 		stackOpts = append(stackOpts, challengemanager.StackWithAPIEnabled(apiAddr, apiDBPath))
 	}
+	if !config.AutoDeposit {
+		stackOpts = append(stackOpts, challengemanager.StackWithoutAutoDeposit())
+	}
+	if !config.AutoIncreaseAllowance {
+		stackOpts = append(stackOpts, challengemanager.StackWithoutAutoAllowanceApproval())
```

Comment:
Why isn't this code also for the assertion chain? Does the assertion chain not do auto allowance? Same thing for the delegated staking, maybe I'm just not sure what's the responsibility of the challenge manager vs assertion chain.

Reply:
The challenge manager spins up something called an assertion manager, and it is the assertion manager that does the auto allowance approval, auto deposit, and delegated staking on startup. The `assertion chain` is an abstraction over the Go bindings to Solidity which also do auto-deposit each time it makes a move related to the stake token. Perhaps we should unify all those responsibilities under the assertion chain?

Reply:
No need for now, makes sense and LGTM aside from the presumably accidental testnode change